hazelcast-jclouds/src/main/java/com/hazelcast/jclouds/ComputeServiceBuilder.java

```diff
@@ -218,17 +251,18 @@ public boolean apply(ComputeMetadata nodeMetadata) {
                     return false;
                 }
                 for (AbstractMap.SimpleImmutableEntry entry: tagPairs) {
-                    if (!nodeMetadata.getUserMetadata().get(entry.getKey()).
-                            equals(entry.getValue())) {
-                        return false;
+                    String value = nodeMetadata.getUserMetadata().get(entry.getKey());
+                    if (value != null && value.equals(entry.getValue())) {
```

Comment:
![](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/images/Tag_Example.png)

The picture above says we need AND operation here. Otherwise, following config will return both machines instead of `production` machine.

```
<property name="tag-keys">owner,Stack</property>
 <property name="tag-values">DbAdmin,Production</property>
```